File: solution/344ReverseString/344ReverseString.cpp
#include <iostream>
using namespace std;

class Solution {
public:
  string reverseString(string s) {
    int lo = 0;
    int hi = int(s.size()) - 1;
    while (lo < hi) {
      swap(s[lo++], s[hi--]);
    }
    return s;
  }
};

int main() { return 0; }

File: src/344ReverseString/344ReverseString.rs
impl Solution {
    pub fn reverse_string(s: String) -> String {
        let n = s.len();
        if n <= 1 {return s;}
        let mut chars = s.into_bytes();    
        let mut i = 0;
        let mut j = n-1;
        while i<j {
            chars.swap(i,j);
            i+=1;
            j-=1;
        }
        String::from_utf8(chars).unwrap()
    }
}
